File: Week3/Code/get_TreeHeight.R
#!/usr/bin/Rscript
# Author - Jacob Griffiths, jacob.griffiths18@imperial.ac.uk
# Date - Oct 2018library(tools)

# This function calculates heights of trees given distance of each tree
# from its base and angle to its top, using the trigonometric formula
#
# height = distance * tan(radians)
#
# ARGUMENTS - any .csv file with the following data:
# degrees:  The angle of elevation of tree
# distance: The distance from base of tree (e.g., metres)
#
# OUTPUT
# The heights of the tree, same units as "distance"

library(tools)

# Using commandArgs to import and .csv file that is specified
args <- commandArgs(trailingOnly = TRUE)
file_in <- args[1]
MyData <- read.csv(file = file_in, header = TRUE) 

for (i in MyData){ #i here just means for the 'ith' row
        radians <- MyData$Angle.degrees * pi / 180 # Referencing columns directly
        height <- MyData$Distance.m * tan(radians)
        Tree.Height.m <- c(height) # Make a new vector called Tree,height.m
}

tree.heights <- data.frame(MyData, Tree.Height.m)

# Stripping file extension of input file
file_out <- basename(file_path_sans_ext(file_in))

# Writing new .csv with results in
write.csv(tree.heights, paste0("../Results/", file_out, "_treeheights", ".csv"))

File: Week3/Code/get_TreeHeight.py
#!/usr/bin/env python3

""" A program that identifies oaks from a list of taxa. 
    A bug was initially present but has been removed and 
    doctests have been added to test the functions.  """

__appname__ = 'oaks_debugme.py'
__author__ = 'Jacob Griffiths (jacob.griffiths18@imperial.ac.uk)'
__version__ = '0.0.1'

import pandas as pd
import scipy as sc
import math
import os
import sys

pd.set_option("display.precision", 13)
MyData = pd.read_csv(sys.argv[1], sep=',', float_precision='round_trip')
MyDF = pd.DataFrame(data=MyData)

MyDF["Tree.Height.m"] = MyDF["Angle.degrees"].apply(lambda x: math.tan(x * (math.pi / 180)))
MyDF['Tree.Height.m'] = MyDF['Tree.Height.m']*MyDF['Distance.m']


# Stripping file extension of input file
name = os.path.splitext(os.path.basename(sys.argv[1]))[0]

# Writing new .csv with results in
MyDF.to_csv("../Results/" + name + "_treeheights_py.csv", float_format="%.13f")